### ISSUE TITLE
Fix dsCheckData converting matrices to data structure in MATLAB 2017a

### ISSUE THREAD

**Issue by jsherfey** [DynaSim/DynaSim#427]
MATLAB 2017a no longer supports `data.data=data` where `data` is initially a matrix; this is done in dsCheckData (line 48) to convert a data matrix into a DynaSim data structure. The error message is "Field assignment to a non-structure array object." The error can be generated by executing `dsCheckData(rand(100,10))` in MATLAB 2017a. In MATLAB 2014a, this produces:

```
dsCheckData(rand(100,10))
ans = 
       labels: {'pop1_data'  'time'}
         time: [100x1 double]
        model: [1x1 struct]
    pop1_data: [100x10 double]
```

This problem was discovered by running the example code for `getPopRhythmGating`.

**Comment by kupiqu** [comment]
I guess in the end this says a bit in favor of Matlab, as overwriting matrices with struct variables may be error prone, i.e. sth done not voluntarily

**Comment by erik-roberts** [comment]
so it just needs to be fixed with a temp variable right?
```matlab
tempData = data;
clear data
data.data=tempData;
```

**Comment by jsherfey** [comment]
Yes, and then to be tested in MATLAB 2016a or 2017a to make sure the rest
of dsCheckData is compatible with the newer versions.
